python/MooseDocs/extensions/MooseCarousel.py

```diff
@@ -23,6 +24,24 @@ class MooseCarousel(BlockProcessor, MooseCommonExtension):
     # they need to have different ids
     MATCHES_FOUND = 0
 
+    def __init__(self, parser, root=None, **kwargs):
+      MooseCommonExtension.__init__(self)
+      BlockProcessor.__init__(self, parser, **kwargs)
+
+      self._root = os.path.join(root, 'docs/media')
+
+      # The default settings
+      self._settings = {'caption'  : None,
+                        'interval' : None,
+                        'pause'    : None,
+                        'wrap'     : None,
+                        'keyboard' : None}
+
+      # We need a way to limit CSS to elements that we know causes issues.
+      # example: to ignore overflow-y to only div elements
+      # self.invalid_css = { 'div' : ['overflow-y'] }
+      self._invalid_css = {}
```

Comment:
The empty dict() should be provided by the MooseCommonExtension, just like self._settings. If the derived class needs something here, then it populates the dict.